lib/autoupdate/agent/updater.go

```diff
@@ -348,18 +360,79 @@ func (u *Updater) Install(ctx context.Context, override OverrideConfig) error {
 	// Only write the configuration file if the initial update succeeds.
 	// Note: skip_version is never set on failed enable, only failed update.
 
-	if err := writeConfig(u.ConfigPath, cfg); err != nil {
+	if err := writeConfig(u.UpdateConfigPath, cfg); err != nil {
 		return trace.Wrap(err, "failed to write %s", updateConfigName)
 	}
 	u.Log.InfoContext(ctx, "Configuration updated.")
 	return trace.Wrap(u.notices(ctx))
 }
 
+// proxyAddrConfig contains potential proxy server addresses from teleport.yaml.
+type proxyAddrConfig struct {
+	AuthServers []string `yaml:"auth_servers"`
+	AuthServer  string   `yaml:"auth_server"`
+	ProxyServer string   `yaml:"proxy_server"`
+}
+
+// findAgentProxy finds a proxy in teleport.yaml if not specified or set in update configuration.
+// Note that any implicitly defaulted port in teleport.yaml is explicitly defaulted (to 3080) by this method.
+func (u *Updater) findAgentProxy(ctx context.Context) string {
+	f, err := libutils.OpenFileAllowingUnsafeLinks(u.TeleportConfigPath)
+	if err != nil {
+		u.Log.DebugContext(ctx, "Unable to open Teleport config to read proxy", "config", u.TeleportConfigPath, errorKey, err)
+		return ""
+	}
+	defer f.Close()
+	var cfg proxyAddrConfig
+	if err := yaml.NewDecoder(f).Decode(&cfg); err != nil {
+		u.Log.DebugContext(ctx, "Unable to parse Teleport config to read proxy", "config", u.TeleportConfigPath, errorKey, err)
+		return ""
+	}
+	var addr string
+	var port int
+	switch {
```

Comment:
@hugoShaka Reading through `lib/service/service.go` and the config parsing logic, I think this will return the best candidate address without attempting to parse anything else in the file that could change.